File: backend/src/main/java/com/mysite/knitly/domain/order/controller/OrderController.java
package com.mysite.knitly.domain.order.controller;

import com.mysite.knitly.domain.order.dto.OrderCreateRequest;
import com.mysite.knitly.domain.order.dto.OrderCreateResponse;
import com.mysite.knitly.domain.order.service.OrderFacade;
import com.mysite.knitly.domain.order.service.OrderService;
import com.mysite.knitly.domain.user.entity.User;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.UUID;

@RestController
@RequiredArgsConstructor
@RequestMapping("/orders")
public class OrderController {

    private final OrderFacade orderFacade;
    @PostMapping
    public ResponseEntity<OrderCreateResponse> createOrder(
            @AuthenticationPrincipal User user,
            @RequestBody @Valid OrderCreateRequest request
    ) {
        // Facade의 메서드를 호출하도록 변경
        OrderCreateResponse response = orderFacade.createOrderWithLock(user, request);
        return ResponseEntity.ok(response);
    }
}

File: backend/src/main/kotlin/com/mysite/knitly/domain/order/controller/OrderController.kt
package com.mysite.knitly.domain.order.controller

import com.mysite.knitly.domain.order.dto.OrderCreateRequest
import com.mysite.knitly.domain.order.dto.OrderCreateResponse
import com.mysite.knitly.domain.order.service.OrderFacade
import com.mysite.knitly.domain.user.entity.User
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/orders")
class OrderController(
    private val orderFacade: OrderFacade
) {
    @PostMapping
    fun createOrder(
        @AuthenticationPrincipal user: User,
        @RequestBody @Valid request: OrderCreateRequest
    ): ResponseEntity<OrderCreateResponse> {
        val response = orderFacade.createOrderWithLock(user, request)
        return ResponseEntity.ok(response)
    }
}